### PR TITLE
TRUNK-4667 Move on to Spring Data JPA - WORK IN PROGRESS

### PR DESCRIPTION
**Pull request by AdamGrzybkowski**
## Related Issue

see https://issues.openmrs.org/browse/TRUNK-4667

## Checklist:

- [ ] My pull request only contains one single commit.
- [ ] My pull request is based on the latest master branch
  `git pull --rebase upstream master`.
- [ ] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.
- [ ] My code follows the code style of this project.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: api/src/main/java/org/openmrs/config/DatabaseConfig.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+@PropertySource("classpath:hibernate.default.properties")
+public class DatabaseConfig {
+
+    @Bean(name = "entityManagerFactory")
+    LocalContainerEntityManagerFactoryBean entityManagerFactory(DataSource dataSource, Environment env) {
+        LocalContainerEntityManagerFactoryBean entityManagerFactoryBean = new LocalContainerEntityManagerFactoryBean();
+        entityManagerFactoryBean.setDataSource(dataSource);
+        entityManagerFactoryBean.setJpaVendorAdapter(new HibernateJpaVendorAdapter());
+        entityManagerFactoryBean.setMappingResources("org/openmrs/api/db/hibernate/Allergy.hbm.xml",
+                "org/openmrs/api/db/hibernate/AllergyReaction.hbm.xml",
```
Comment (AdamGrzybkowski) [inline api/src/main/java/org/openmrs/config/DatabaseConfig.java]: This is only temporary but i couldn't find a way to set mappings from hibernate.cfg.xml file directly

File: api/src/main/resources/applicationContext-service.xml [hunk 2]
```diff
@@ -582,20 +582,8 @@
...
-		</property>
-		<property name="mappingJarLocations">
-			<ref bean="mappingJarResources" />
-		</property>
-        <property name="packagesToScan">
-            <list>
-                <value>org.openmrs</value>
-            </list>
-            </property>
+	<bean id="sessionFactory" class="org.springframework.orm.jpa.vendor.HibernateJpaSessionFactoryBean">
```
Comment (AdamGrzybkowski) [inline api/src/main/resources/applicationContext-service.xml]: I had to edit this to make it work together with JPA according to http://stackoverflow.com/questions/21671731/simultaneous-use-of-hibernate-and-spring-data-jpa

**Comment by AdamGrzybkowski** [comment]
@rkorytkowski I'm kind of stuck with this. I'm getting this error now http://pastebin.com/isU4CSVB. I've left some comments, to explain why I did what I did :)

File: api/pom.xml [hunk 3]
```diff
@@ -88,6 +88,11 @@
          <artifactId>spring-context-support</artifactId>
       </dependency>
       <dependency>
+         <groupId>org.springframework.data</groupId>
+         <artifactId>spring-data-jpa</artifactId>
+         <version>1.9.1.RELEASE</version>
```
Comment (AdamGrzybkowski) [inline api/pom.xml]: Not the latest version,because of compatibility issue with Spring. The latest one requires Spring 4.2.8.RELEASE

**Comment by rkorytkowski** [comment]
Could you please say what is the issue? Are you getting any stacktraces?

**Comment by rkorytkowski** [comment]
I don't know what's the issue off hand. I'll have to run the code. I may have some time later today. Please continue working on other issues.

**Comment by dkayiwa** [comment]
@AdamGrzybkowski did you get to fix the error you mentioned above?

**Pull request closed without merging**